File: fm-domain/src/main/java/io/zeitwert/fm/oe/model/ObjUserFMRepository.java
package io.zeitwert.fm.oe.model;

import java.util.Optional;

import org.springframework.security.crypto.password.PasswordEncoder;

import io.dddrive.core.obj.model.ObjRepository;
import io.zeitwert.fm.dms.model.ObjDocumentRepository;

public interface ObjUserFMRepository extends ObjRepository<ObjUserFM> {

	PasswordEncoder getPasswordEncoder();

	ObjDocumentRepository getDocumentRepository();

	Optional<ObjUserFM> getByEmail(String email);

	boolean isAppAdmin(ObjUserFM user);

	boolean isAdmin(ObjUserFM user);

}

File: fm-domain/src/main/java/io/zeitwert/fm/oe/model/ObjUserFMRepository.kt
package io.zeitwert.fm.oe.model

import io.zeitwert.fm.dms.model.ObjDocumentRepository
import io.zeitwert.fm.obj.model.FMObjRepository
import org.springframework.security.crypto.password.PasswordEncoder
import java.util.*

interface ObjUserFMRepository : FMObjRepository<ObjUserFM> {

	val passwordEncoder: PasswordEncoder

	val documentRepository: ObjDocumentRepository

	fun getByEmail(email: String): Optional<ObjUserFM>

	fun isAppAdmin(user: ObjUserFM): Boolean

	fun isAdmin(user: ObjUserFM): Boolean

}
